src/transformers/models/whisper/generation_whisper.py

```diff
@@ -727,7 +729,9 @@ def generate(
             if (prompt_ids is not None and generation_config.prompt_condition_type == "first-segment")
             else current_segments
         )
-        sequences = _pad_to_max_length(final_segments, generation_config.pad_token_id, padding="right")
+        sequences = _pad_to_max_length(
+            final_segments, generation_config.pad_token_id, device=self.device, padding="right"
```

Comment:
Is self.device guaranteed to be the same device as `final_segments`? What happens if you load the model using `device_map="auto"` in a multi-gpu setting?

Reply:
Not sure how `auto` device works, but on my set-up it works well, adding some tests rn!